src/components/_global/BalAsset/BalAsset.vue

```diff
@@ -1,96 +1,111 @@
 <template>
-  <img
-    v-if="iconSRC && !error"
-    :src="iconSRC"
+  <component
+    class="bal-asset rounded-full inline-block leading-none shadow-sm"
     :style="{
       width: `${size}px`,
-      height: `${size}px`,
-      background: 'white'
+      height: `${size}px`
     }"
-    @error="error = true"
-    class="rounded-full inline-block leading-none shadow-sm"
-  />
-  <Avatar v-else-if="!!address" :address="address" :size="size" />
-  <div
-    v-else
-    class="rounded-full inline-block leading-none shadow-sm overflow-visible"
-    :style="{
-      width: `${size}px`,
-      height: `${size}px`,
-      background: `${noIconColor} !important`
-    }"
-  />
+    :is="rootElement"
+    v-bind="rootElementAttrs"
+  >
+    <img
+      v-if="iconSRC && !error"
+      :src="iconSRC"
+      :style="{
+        background: 'white'
```

Comment:
@pkattera Is it ok to keep the image background as `'white'`, or should it be inherited from theme?

Reply:
@sssmi I agree with @insulineru that it's best to use Tailwind styles when possible. So `bg-white` is best.